### PR TITLE
Fix #811: Unit test InputFormatBuilder.java

### LINKED ISSUE #811: Missing tests in hadoop-mapreduce
Tests should be added for the following options (which probably aren't even tested in core.mapred*):
- InputFormatBuilder -> classLoaderContext
- InputFormatBuilder -> executionHints
- OutputFormatBuilder -> simulationMode

### PR DESCRIPTION
**Pull request by elinaawise**
Unit test classLoaderContext and executionHints() functions

### CODE REVIEW

**Comment by ivakegg** [comment]
There appear to be many more methods in the InputFormatBuilderImpl that could be tested.  What was the initial goal here?

**Comment by ivakegg** [comment]
> There appear to be many more methods in the InputFormatBuilderImpl that could be tested. What was the initial goal here?

I see that most of the other methods are tested in other places, so I am good with the brevity of these tests.

**Comment by ctubbsii** [comment]
Anybody know the status of this PR?

**Comment by elinaawise** [comment]
> Anybody know the status of this PR?

Ivan reviewed this test last week.  Believe it just needs to be merged.

**Comment by ctubbsii** [comment]
@elinaawise Oh, the PR is still marked with "change requested" from him, so I didn't know if he hadn't changed that because there was more to do. I can take a look myself today.

**Comment by elinaawise** [comment]
> @elinaawise Oh, the PR is still marked with "change requested" from him, so I didn't know if he hadn't changed that because there was more to do. I can take a look myself today.

I see it now - Ivan requested I add comments in 2 areas of the test.  I'll get these changes in today.

File: hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoopImpl/mapreduce/InputFormatBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  @Test
+  public void testInputFormatBuilderImplTest_ExecuteHints() {
+    Map<String,String> hints = ImmutableMap.<String,String>builder().put("key1", "value1")
+        .put("key2", "value2").put("key3", "value3").build();
+
+    formatBuilderTest.executionHints(hints);
+
+    Map<String,String> executionHints = formatBuilderTest.getExecutionHints();
+    assertEquals(executionHints.toString(), hints.toString());
```
[the remark this answers is not in the document]
Comment (ivakegg) [reply hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoopImpl/mapreduce/InputFormatBuilderTest.java]: In an asserts statement, the expected values is the first argument.  Please switch the order of the arguments.

File: hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoopImpl/mapreduce/InputFormatBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  public void testInputFormatBuilderImplTest_ExecuteHints() {
+    SortedMap<String,String> hints = new TreeMap<>();
+    hints.put("key1", "value1");
+    hints.put("key2", "value2");
+    hints.put("key3", "value3");
+
+    formatBuilderTest.executionHints(hints);
+
+    SortedMap<String,String> executionHints = formatBuilderTest.getExecutionHints();
+    assertEquals(executionHints.toString(), hints.toString());
```
Comment (ivakegg) [inline hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoopImpl/mapreduce/InputFormatBuilderTest.java]: In an asserts statement, the expected values is the first argument. Please switch the order of the arguments.

**Comment by ctubbsii** [comment]
Thanks, @elinaawise ; I added a commit to address Ivan's last comment, and also to fix some of the generics warnings, and will merge this in.

**Pull request merged**